modules/ducktests/tests/ignitetest/tests/discovery_test.py

```diff
@@ -120,8 +148,20 @@ def __basic_test__(self, version, with_zk=False):
         data = {'Ignite cluster start time (s)': self.monotonic() - start}
         self.stage("Topology is ready")
 
-        # Node failure detection
-        fail_node, survived_node = self.choose_random_node_to_kill(self.servers)
+        if nodes_to_kill > self.servers.num_nodes - 1:
+            raise Exception("Too many nodes to kill: " + str(nodes_to_kill))
+
+        if with_zk:
+            node_chooser = lambda nodes: random.sample(nodes, nodes_to_kill)
+        elif coordinator:
+            node_chooser = lambda nodes: \
+                next(node for node in nodes if node.discovery_info().node_id == nodes[0].discovery_info().coordinator)
+        else:
+            node_chooser = lambda nodes: \
+                random.sample([n for n in self.servers.nodes if n.discovery_info().node_id
+                               != self.servers.nodes[0].discovery_info().coordinator], nodes_to_kill)
```

Comment:
I suppose, that != should remain on previous line

Reply:
Fixed '!='
As for the lambdas, only 2 left. Think it is ok now.